server/game/cards/characters/02/unswornapprentice.js

```diff
@@ -0,0 +1,42 @@
+const _ = require('underscore');
+const DrawCard = require('../../../drawcard.js');
+
+class UnswornApprentice extends DrawCard {
+    setupCardAbilities(ability) {
+        this.action({
+            title: 'Gain Icon',
+            method: 'gainIcon',
```

Comment:
Small thing but you should restrict the phase here using
```
phase: 'challenge',
```
rather than checking it and returning in `gainIcon`.